pkg/template/template_test.go

```diff
@@ -177,6 +177,59 @@ func TestParameterGenerators(t *testing.T) {
 	}
 }
 
+func TestProcessValue(t *testing.T) {
+	var template api.Template
+	if err := runtime.DecodeInto(kapi.Codecs.UniversalDecoder(), []byte(`{
+		"kind":"Template", "apiVersion":"v1",
+		"objects": [
+			{
+				"kind": "Service", "apiVersion": "v${VALUE}",
+				"metadata": {
+					"labels": {
+						"key1": "${VALUE}",
+						"key2": "$${VALUE}",
+						"s1_s1": "${STRING_1}_${STRING_1}",
+						"s1_s2": "${STRING_1}_${STRING_2}",
+						"i1": "${{INT_1}}",
```

Comment:
> Create test conditions for the "error" case is described above

not sure which case you're referring to, but there are three tests for "invalid" ${{KEY}} substitutions ("a${{KEY}}", "${{KEY}}a", and "${{INVALID_PARAMETER}}", all of which ensure that the value isn't touched in those cases (backwards compatible behavior).

> Also bad json substitution error case

There is no "bad json substitution" scenario.  if you use the ${{KEY}} syntax w/ a random string parameter value(non-json value), you end up with a quoted string because we fail to decode it and then use it as a direct string value instead.  that's tested by the "quoted_string" key test.  (ensures that if you use the ${{KEY}} syntax but the parameter value is not an int/bool, you end up with a quoted string containing the parameter value)

I don't think there's any case where we would explicitly error based on the substitution logic, we will either:  do a string substitution instead of an int/boolean substitution, or we will do no substitution (leaving a string value with a ${{FOO}} reference embedded).

Those things might result in an invalid api object, but we've never done checking for that sort of thing, the template processing is api object agnostic.

Guessing there is still a scenario you're thinking of that i'm missing but i don't see it at the moment.